front/src/components/login/register.jsx

```diff
@@ -0,0 +1,189 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import { FormErrors } from "./FormErrors";
+import {authenticationServise} from "../helpers/athentication";
+export class Register extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: "",
+      password: "",
+      formErrors: {
+        username: "",
+        password: "",
+        email: "",
+        isChecked: false,
+        usernameValid: false,
+        emailValid: false,
+        passwordValid: false,
+        formValid: false
+      }
+    };
+    this.handleUserInput = this.handleUserInput.bind(this);
+    this.handleChecked = this.handleChecked.bind(this);
+  }
+  handleUserInput = e => {
+    const name = e.target.name;
+    const value = e.target.value;
+    this.setState({ [name]: value }, () => {
+      this.validateField(name, value);
+    });
+  };
+  handleChecked(e) {
+    this.setState({ isChecked: !this.state.isChecked });
+    console.log(this.state.isChecked);
+  }
+  validateField(fieldName, value) {
+    let fieldValidationErrors = this.state.formErrors;
+    let usernameValid = this.state.usernameValid;
+    let passwordValid = this.state.passwordValid;
+    let emailValid = this.state.emailValid;
+
+    switch (fieldName) {
+      case "email":
+        emailValid = value.match(
+          /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
+        );
+        fieldValidationErrors.email = emailValid ? "" : " неправильный ";
+        break;
+      case "username":
```

Comment:
You see how you're duplicating code from the other part of the application here as well, what if the requirements change(like, username can't be shorter than 3 symbols). You'll have to remember to change it in both instances.